### ISSUE TITLE
TransportServer: SNAT "self" / "bigip"

### ISSUE THREAD

**Issue by chen23** [F5Networks/k8s-bigip-ctlr#1670]
#### Title
Allow a snat value of "self" and/or "bigip"

#### Description
Customer currently uses a SNAT pool that matches the IP address of the virtual server.  Allowing the SNAT value in AS3 to be set to "self" would mirror this configuration.

#### Actual Problem
Only NONE/Auto are supported SNAT types (according to docs)

#### Solution Proposed
Allow "Self" as another option (allowing "bigip" to reference an existing SNAT pool would also be desirable)

#### Alternatives
One could attach an iRule to do the same, but that would not be as nice.

**Comment by chen23** [comment]
currently you get the following error if you try to specify "self"
```
2021/01/27 16:53:27 [INFO] BIGIP is serving with AS3 version : 3.24.0-5
I0127 16:53:27.751189       1 shared_informer.go:204] Caches are synced for F5 CIS CRD Controller
2021/01/27 16:53:28 [INFO] [2021-01-27 16:53:28,510 __main__ INFO] entering inotify loop to watch /tmp/k8s-bigip-ctlr.config023234891/config.json
2021/01/27 18:29:23 [ERROR] [AS3] Big-IP Responded with code: 422
2021/01/27 18:29:23 [ERROR] [AS3] Raw response from Big-IP: map[code:422 declarationFullId: errors:[/cluster1/Shared/crd_10_1_10_82_8080/snat/bigip: should match format "f5bigip"] message:declaration is invalid]
```

**Comment by trinaths** [comment]
@chen23 This feature is available with CIS v2.3. Closing this issue.
